File: apps/web/src/lib/api/analytics.ts
import { apiClient } from './client';

export interface AnalyticsSummary {
  total_drafts_generated: number;
  drafts_approved: number;
  drafts_rejected: number;
  drafts_published: number;
  approval_rate: number;
  avg_confidence: number;
  period: string;
  period_start: string;
  period_end: string;
}

export interface TopicMetric {
  topic: string;
  count: number;
  approval_rate: number;
}

export interface TopicAnalytics {
  topics: TopicMetric[];
  period: string;
}

export interface FormatMetric {
  format: string;
  count: number;
  approval_rate: number;
}

export interface FormatAnalytics {
  formats: FormatMetric[];
  period: string;
}

export const analyticsApi = {
  getSummary: async (profileId: string, period: string = '30d'): Promise<AnalyticsSummary> => {
    const response = await apiClient.get('/analytics/summary', {
      params: { profile_id: profileId, period },
    });
    return response.data;
  },

  getTopics: async (profileId: string, period: string = '30d'): Promise<TopicAnalytics> => {
    const response = await apiClient.get('/analytics/topics', {
      params: { profile_id: profileId, period },
    });
    return response.data;
  },

  getFormats: async (profileId: string, period: string = '30d'): Promise<FormatAnalytics> => {
    const response = await apiClient.get('/analytics/formats', {
      params: { profile_id: profileId, period },
    });
    return response.data;
  },
};

File: apps/api/app/schemas/analytics.py
"""Analytics schemas."""
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class AnalyticsSummary(BaseModel):
    """Schema for analytics summary."""

    total_drafts_generated: int = 0
    drafts_approved: int = 0
    drafts_rejected: int = 0
    drafts_published: int = 0
    approval_rate: float = 0.0
    avg_confidence: float = 0.0
    period: str = "30d"
    period_start: datetime
    period_end: datetime


class TopicMetric(BaseModel):
    """Schema for topic metrics."""

    topic: str
    count: int
    approval_rate: float


class TopicAnalytics(BaseModel):
    """Schema for topic analytics."""

    topics: list[TopicMetric] = Field(default_factory=list)
    period: str = "30d"


class FormatMetric(BaseModel):
    """Schema for format metrics."""

    format: str
    count: int
    approval_rate: float


class FormatAnalytics(BaseModel):
    """Schema for format analytics."""

    formats: list[FormatMetric] = Field(default_factory=list)
    period: str = "30d"


class AgentSuggestion(BaseModel):
    """Schema for agent tuning suggestion."""

    suggestion_type: str
    description: str
    impact: str
    action: Optional[str] = None


class WeeklyDigest(BaseModel):
    """Schema for weekly analytics digest."""

    summary: AnalyticsSummary
    top_topics: list[TopicMetric] = Field(default_factory=list)
    top_formats: list[FormatMetric] = Field(default_factory=list)
    suggestions: list[AgentSuggestion] = Field(default_factory=list)
    generated_at: datetime
